### PR TITLE
KAFKA-6871: KStreams Scala API: incorrect Javadocs and misleading parameter name

### PR DESCRIPTION
**Pull request by miguno**
See PR title. This is a minor fix.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala [hunk 1]
```diff
@@ -261,30 +261,30 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
...
-   * A `Transformer` (provided by the given `TransformerSupplier`) is applied to each input record and
-   * computes zero or more output records. In order to assign a state, the state must be created and registered 
-   * beforehand via stores added via `addStateStore` or `addGlobalStore` before they can be connected to the `Transformer` 
+   * A `Transformer` is applied to each input record and computes zero or more output records. In order to assign a
+   * state, the state must be created and registered beforehand via stores added via `addStateStore` or `addGlobalStore`
+   * before they can be connected to the `Transformer`
    *
-   * @param transformerSupplier a instance of `TransformerSupplier` that generates a `Transformer`
+   * @param transformer the `Transformer` instance
    * @param stateStoreNames     the names of the state stores used by the processor
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala:269]: nit: align idention

**Review by mjsax — APPROVED** [review]
LGTM.

**Comment by mjsax** [comment]
\cc @bbejeck @vvcephei @debasishg

**Comment by debasishg** [comment]
LGTM

**Review by guozhangwang — APPROVED** [review]

**Review by bbejeck — COMMENTED** [review]
Thanks, @mjsax LGTM

**Comment by mjsax** [comment]
@bbejeck It's not my PR -- thanks goes to @miguno :)

**Comment by mjsax** [comment]
Thanks @miguno ! Merged to `trunk`

**Comment by bbejeck** [comment]
@miguno mea-culpa!  Thanks for the PR

**Pull request merged**